### PR TITLE
VZ-4435: POKO: Implement acceptance tests for all standard Kubernetes workloads

### PR DESCRIPTION
**Pull request by bobdonat**
# Description

The pull request adds acceptance tests to verify Prometheus metrics are scraped for the standard Kubernetes workloads.  The workloads tests are deployments, pods, replicasets, and statefulsets.

# Checklist 

As the author of this PR, I have:

- [x] Checked that I included or updated copyright and license notices in all files that I altered
- [ ] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [x] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: tests/e2e/scrapegenerator/helidon-deployment/helidon_deployment_suite_test.go [hunk 1]
```diff
@@ -1,7 +1,7 @@
 // Copyright (c) 2021, 2022, Oracle and/or its affiliates.
 // Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
 
-package helidonscrapegenerator
+package scrapedeploymentworkload
```
Comment (kminder) [inline tests/e2e/scrapegenerator/helidon-deployment/helidon_deployment_suite_test.go:4]: Is it time for me to hate on 'scrape' and 'scrapegenerator' in all these names?

Comment (markxnelson) [reply tests/e2e/scrapegenerator/helidon-deployment/helidon_deployment_suite_test.go:4]: yes, please

**Review by markxnelson — APPROVED** [review]
LGTM

File: tests/e2e/scrapegenerator/helidon-statefulset/helidon_statefulset_test.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+	"github.com/verrazzano/verrazzano/tests/e2e/pkg"
+	"github.com/verrazzano/verrazzano/tests/e2e/scrapegenerator"
+)
+
+const (
+	longWaitTimeout      = 15 * time.Minute
+	longPollingInterval  = 20 * time.Second
+	namespace            = "hello-helidon-namespace"
+	applicationPodPrefix = "hello-helidon-statefulset-"
+	yamlPath             = "application-operator/internal/app/resources/workloads/hello-helidon-statefulset.yaml"
```
Comment (kminder) [inline tests/e2e/scrapegenerator/helidon-statefulset/helidon_statefulset_test.go:21]: I don't see the YAML files for the other workload types in this PR.  Were they already merged as part of some other change?

Comment (bobdonat) [reply tests/e2e/scrapegenerator/helidon-statefulset/helidon_statefulset_test.go:21]: yaml files are existed and had already been merged

File: tests/e2e/scrapegenerator/helidon-deployment/helidon_deployment_suite_test.go (hunk 1 above)
Comment (bobdonat) [reply tests/e2e/scrapegenerator/helidon-deployment/helidon_deployment_suite_test.go:4]: i will get rid of scrape

File: tests/e2e/scrapegenerator/helidon-replicaset/helidon_replicaset_test.go [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+var clusterDump = pkg.NewClusterDumpWrapper()
+var _ = clusterDump.AfterEach(func() {}) // Dump cluster if spec fails
+var _ = clusterDump.AfterSuite(func() {  // Dump cluster if aftersuite fails
+	scrapegenerator.UndeployApplication(namespace, yamlPath)
+})
+
+var _ = t.AfterEach(func() {})
+
+var _ = t.Describe("Verify application.", func() {
```
Comment (kminder) [inline tests/e2e/scrapegenerator/helidon-replicaset/helidon_replicaset_test.go:40]: For the OAM MetricsTrait there are tests that check to ensure the Prometheus config was updated and cleaned up.
https://github.com/verrazzano/verrazzano/blob/master/tests/e2e/metrics/deploymetrics/deploymetrics_test.go#L81
Sort of an implementation detail but checking it did catch some issues.

Comment (bobdonat) [reply tests/e2e/scrapegenerator/helidon-replicaset/helidon_replicaset_test.go]: added this check after undeploying workload resource

File: application-operator/internal/app/resources/workloads/hello-helidon-pod.yaml [hunk 4]
```diff
@@ -4,7 +4,7 @@
 apiVersion: v1
 kind: Pod
 metadata:
-  name: hello-helidon-deployment
+  name: hello-helidon-pod
```
Comment (pfmackin) [inline application-operator/internal/app/resources/workloads/hello-helidon-pod.yaml:7]: putting the resource kind in the name is not needed.  I know we do this for other apps but we shouldn't.

File: tests/e2e/metricsbinding/helidon-pod/helidon_pod_test.go [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+	yamlPath             = "application-operator/internal/app/resources/workloads/hello-helidon-pod.yaml"
+	promConfigJobName    = "hello-helidon-namespace_hello-helidon-pod_v1_Pod"
+)
+
+var t = framework.NewTestFramework("podworkload")
+
+var _ = t.BeforeSuite(func() {
+	start := time.Now()
+	metricsbinding.DeployApplication(namespace, yamlPath)
+	metrics.Emit(t.Metrics.With("deployment_elapsed_time", time.Since(start).Milliseconds()))
```
Comment (pfmackin) [inline tests/e2e/metricsbinding/helidon-pod/helidon_pod_test.go:30]: Doesn't this time include non-deployment time, i.e. the time to script metrics?

Comment (bobdonat) [reply tests/e2e/metricsbinding/helidon-pod/helidon_pod_test.go:30]: Its the time to create the namespace and deploy the application

File: application-operator/internal/app/resources/workloads/hello-helidon-pod.yaml (hunk 4 above)
Comment (bobdonat) [reply application-operator/internal/app/resources/workloads/hello-helidon-pod.yaml:7]: yes, it is not needed.  we were trying to distinguish the different kinds of hello-helidon deployments we are testing.  We have pods, deployments, statefulsets, and replicsets

**Review by kminder — APPROVED** [review]
Approved.

**Pull request merged**